File: Python/ExampleTests/common/filter_value_typed.py
from . import json_util as _json
from .filter_value_string import FilterValueString

class FilterValueTyped:
    def __init__(self, value: str = ''):
        self.value = value

    @classmethod
    def from_string_obj(cls, s: FilterValueString) -> 'FilterValueTyped':
        return cls(
            s.value
        )

    def to_json_value(self) -> dict:
        return {
            'value': self.value,
        }

    def to_json(self) -> str:
        return _json.dumps(self.to_json_value())

    @classmethod
    def from_json_value(cls, m: dict) -> 'FilterValueTyped':
        return cls(
            _json.as_str(_json.require(m, 'value'), 'value')
        )

    @classmethod
    def from_json(cls, text: str) -> 'FilterValueTyped':
        return cls.from_json_value(_json.loads(text))

    @staticmethod
    def to_json_list(items) -> str:
        return _json.dumps([item.to_json_value() for item in items])

    @classmethod
    def from_json_list(cls, text: str) -> list:
        raw = _json.as_list(_json.loads(text), 'FilterValueTyped')
        return [cls.from_json_value(e) for e in raw]

    def __str__(self):
        return (f'Value={self.value}')

    def _key(self):
        return (self.value,)

    def __eq__(self, other):
        if not isinstance(other, FilterValueTyped):
            return NotImplemented
        return self._key() == other._key()

    def __hash__(self):
        return hash(self._key())

    def __repr__(self):
        return f'FilterValueTyped({self})'

File: Swift/ExampleTests/common/FilterValueTyped.swift
public struct FilterValueTyped: Equatable, CustomStringConvertible {
    public let value: String

    public init(value: String) {
        self.value = value
    }

    public init(from s: FilterValueString) {
        self.value = s.value
    }

    public func toStringStruct() -> FilterValueString {
        return FilterValueString(
            value: String(describing: value)
        )
    }

    public static func toStringList(_ list: [FilterValueTyped]) -> [FilterValueString] {
        return list.map { $0.toStringStruct() }
    }

    public static func fromStringList(_ list: [FilterValueString]) -> [FilterValueTyped] {
        return list.map { FilterValueTyped(from: $0) }
    }

    public func toJSONValue() -> [String: Any] {
        return [
            "value": value,
        ]
    }

    public func toJSON() throws -> String {
        return try Json.write(toJSONValue())
    }

    public init(fromJSONValue m: [String: Any]) throws {
        self.value = try Json.asString(Json.require(m, "value"), "value")
    }

    public init(fromJSON text: String) throws {
        try self.init(fromJSONValue: Json.parseObject(text))
    }

    public static func toJSONList(_ list: [FilterValueTyped]) throws -> String {
        return try Json.write(list.map { $0.toJSONValue() })
    }

    public static func fromJSONList(_ text: String) throws -> [FilterValueTyped] {
        return try Json.parseArray(text).map {
            try FilterValueTyped(fromJSONValue: Json.asObject($0, "FilterValueTyped"))
        }
    }

    public var description: String {
        return "Value=\(value)"
    }
}
